certified-operators/mongodb-enterprise/mongodb-enterprise.v0.9.0.clusterserviceversion.yaml

```diff
@@ -3,20 +3,70 @@
-  name: mongodboperator.v0.3.2
+  name: mongodboperator.v0.9.0
   namespace: placeholder
   annotations:
     capabilities: Full Lifecycle
     categories: "Database"
     description: The MongoDB Enterprise Kubernetes Operator enables easy deploys of MongoDB into Kubernetes clusters, using our management, monitoring and backup platforms, Ops Manager and Cloud Manager.
     alm-examples: >-
-      [{"apiVersion":"mongodb.com/v1","kind":"MongoDbStandalone","metadata":{"name":"my-standalone","namespace":"mongodb"},"spec":{"version":"4.0.2","persistent":false,"project":"my-project","credentials":"my-credentials"}},{"apiVersion":"mongodb.com/v1","kind":"MongoDbReplicaSet","metadata":{"name":"my-replica-set","namespace":"mongodb"},"spec":{"members":3,"version":"4.0.2","persistent":false,"project":"my-project","credentials":"my-credentials"}},{"apiVersion":"mongodb.com/v1","kind":"MongoDbShardedCluster","metadata":{"name":"my-sharded-cluster","namespace":"mongodb"},"spec":{"shardCount": 2, "mongodsPerShardCount": 3, "mongosCount": 2, "configServerCount": 3,"version":"4.0.2","persistent":false,"project":"my-project","credentials":"my-credentials"}}]
+        [
+            {
+                "apiVersion": "mongodb.com/v1",
+                "kind": "MongoDB",
+                "metadata": {
+                    "name": "my-standalone"
+                },
+                "spec": {
+                    "type": "Standalone",
+                    "version": "4.0.6",
+                    "project": "my-project",
+                    "credentials": "my-credentials",
+                    "persistent": false
+                }
+            },
+            {
+                "apiVersion": "mongodb.com/v1",
+                "kind": "MongoDB",
+                "metadata": {
+                    "name": "my-replica-set"
+                },
+                "spec": {
+                    "type": "ReplicaSet",
+                    "version": "4.0.6",
+                    "project": "my-project",
+                    "credentials": "my-credentials",
+                    "persistent": false,
+                    "members": 3
+                }
+            },
+            {
+                "apiVersion": "mongodb.com/v1",
+                "kind": "MongoDB",
+                "metadata": {
+                    "name": "my-sharded-cluster"
+                },
+                "spec": {
+                    "type": "ShardedCluster",
+                    "version": "4.0.2",
+                    "project": "my-project",
+                    "credentials": "my-credentials",
+                    "persistent": false,
+                    "shardCount": 2,
+                    "mongodsPerShardCount": 3,
+                    "mongosCount": 2,
+                    "configServerCount": 3
+                }
+            }
+        ]
+
 spec:
   displayName: MongoDB
   provider:
     name: 'MongoDB, Inc'
   maturity: stable
-  version: 0.3.2
+  version: 0.9.0
+  replaces: 0.3.2
```

Comment:
Does the Operator migrate these CRDs for you? If it doesn't, does it still understand the old CRDs? If this version replaces an older version, what happens?

Reply:
The operator does not migrate the old CRDs and actually does not know about the old structure. This version is not compatible with the old one. We have documentation in place for customer to migrate their deployments, but it involves manual work.

What would be the right way of proceeding with this change, if this version of the operator does not work with the old one?